### ISSUE TITLE
Function macro syntax

### ISSUE THREAD

**Issue by Darukshock** [SpyglassMC/Spyglass#1696]
[Feature Request]
Hi,
I thought it could be handy having autocompletion & syntax checking when calling a function with inline macro.
__Example__:
Let's say I have the function:
```mcfunction
#> player_motion:internal/summon/crystal
$execute if score $n_full_exp player_motion.internal.math matches 1.. anchored eyes positioned ^ ^ ^$(full_d) run function player_motion:internal/summon/loop
$execute anchored eyes positioned ^ ^ ^$(d) summon end_crystal run damage @s 0
```
And this mcdoc snippet in my workspace:
```mcdoc
/// Setup for spawning crystals
dispatch mcdoc:function["player_motion:internal/summon/crystal"] to struct{
    /// distance from crystal
    d: float,
    /// distance from conditional crystals
    full_d: float
}
```
The idea is that we would get error warning in those scenarios:
1) `function player_motion:internal/summon/crystal {d:4.0}` -> missing argument `full_d`
2) `function player_motion:internal/summon/crystal {d:{foo:"bar"},full_d:3.0}` -> expected a float (hovering over `{foo:"bar"}`)
Also, when hovering over either `d` or `full_d`, we would see the comments written above in the mcdoc, just like how mcdoc currently works.

In conclusion, this would be a really nice tool to:
1) Prevent misspelling mistakes when calling our own functions
2) Enhance librairies that have API functions called with macros, so that users can be certain they put the right macro arguments
3) Accelerate developpement process thanks to autocompletion.

**Comment by misode** [comment]
We have something similar planned in #1070
